### ISSUE TITLE
Auto detect slide title for markdown slides

### ISSUE THREAD

**Issue by thawk** [impress/impress.js#816]
We can use the first non-blank line for markdown title.  For a regular presentation, the first line may be suitable for title.
I think it's better than the `step-X`, and can easy navigation.

**Comment by thawk** [comment]
For example, with the example `examples/markdown/index.html`, in the bottom-right `goto` dropbox, it shows `markdown`/`step-2`/`step-3`/`step-4`....  It's hard to select a correct slide to goto.

To get a meaningful title, we can use the first line of each slide.  I think the first line of a slide should be the most important sentence at most time.   With this, the titles of the above example will be `markdown`/`Markdown.js`/`Styles`/`A blockquote & image`/`Code`/`Slide separator`/`Title slide`...  I think it's much easier to goto the right slide.

-----

For the following example, there will be 3 slides, titled `Slide title`/`Sub-title`/`Another point`.

```markdown
# Slide title

Some text

-----

## Sub-title

* bullet point

## Sub-title2

* some thing

-----

Another point
```

**Comment by henrikingo** [comment]
Ah, thanks for clarifying. This makes sense.

**Comment by thawk** [comment]
Closed by #821
